x/data/server/lookup/table.go

```diff
@@ -0,0 +1,159 @@
+package lookup
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"hash/fnv"
+)
+
+// Table is the interface for interacting with a lookup table.
+type Table interface {
+	// GetOrCreateID is an idempotent method for creating or retrieving an unique
+	// shortened identifier for the provided binary value.
+	GetOrCreateID(store KVStore, value []byte) []byte
+
+	// GetValue returns the binary data (if any) corresponding to the provided shortened identifier.
+	GetValue(store KVStore, id []byte) []byte
```

Comment:
I actually created a separate `KVStore` interface with just `Get` and `Set` in this package so there's no dependency on the SDK's `KVStore`. Having the `KVStore` as a member to the `Table` doesn't work really because it's transactional and we don't want to retain a handle to invalid `KVStore`s

Reply:
Ah, right - in production it will change between the calls. Makes sense.